dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusC.java

```diff
@@ -577,4 +734,17 @@ private Expression assignToReturnRegister(Register register, Expression value) {
         }
         return value;
     }
+
+    private void addMemoryObjectTags(Event event, Expression address) {
+        if (address instanceof Register reg) {
+            MemoryObject object = programBuilder.getLocFromReg(reg);
+            if (object != null) {
+                event.addTags(object.getFeatureTags());
+            }
+        } else if (address instanceof BinaryExpressionBase<?,?> binExpr) {
```

Comment:
Don't check against `BinaryExpressionBase` but rather against the interface `BinaryExpression`. All the `...Base` classes are meant as implementation aids and not as interface (I even gave them a special `NoInterface` attribute to mark this), meaning no code should ever refer to them.
Btw. is there a reason that this is restricted to binary expressions?

Reply:
Updated to all expression. It was I saw in one tests binary expression is used, there is no reason to restrict it to binary expressions XD